src/terms.ml

```diff
@@ -117,6 +119,41 @@ type term =
   ; vars  : (string * int) array
   (** Name and arity of the pattern variables bound in the RHS. *) }
 
+(** Trees are used to efficiently choose a rewriting rule given a list of
+    terms (beginning with a symbol) to be rewritten.  The left hand has side
+    of the rule--being the pattern to be matched for the rule to apply--is
+    spread across the {!cons:Node}s of the tree.  When a {!cons:Leaf} is
+    reached, the target is rewrote to the content of the leaf. *)
+ and tree =
+    Leaf of (term_env, term) Bindlib.mbinder
+  (** Hold the targets of rewriting, right hand side of a rule.  In a
+      {!cons:Leaf}[(t, a)], [t] contains the term--if a switch has been
```

Comment:
I don't understand. The type of Leaf is `(term_env, term) Bindlib.mbinder`, not a pair.

Reply:
The doc has been updated in 1ba9d96